packages/builder/src/registry.ts

```diff
@@ -402,6 +443,18 @@ export class OnChainRegistry extends CannonRegistry {
     return [await this._publishPackages(packageDatas)];
   }
 
+  async unpublish(packagesNames: string[], chainId: number, url: string, metaUrl?: string): Promise<string[]> {
+    console.log(bold(blueBright('\nUnpublishing package to the registry on-chain...\n')));
+    const packageData = this._preparePackageData(packagesNames, chainId, url, metaUrl);
+    return [await this._unpublishPackages([packageData])];
+  }
+
+  async unpublishMany(toUnpublish: { name: string[]; chainId: number; url: string; metaUrl?: string }[]): Promise<string[]> {
+    console.log(bold(blueBright('\nUnpublishing packages to the registry on-chain...\n')));
+    const packageDatas = toUnpublish.map((p) => this._preparePackageData(p.name, p.chainId, p.url, p.metaUrl));
+    return [await this._unpublishPackages(packageDatas)];
+  }
```

Comment:
Looks like this fn was added to maintain consistency with `publish` & `publishMany`, but is not really necessary right now, because we always unpublish several tags but from the same packageName. So, this should be removed just to avoid extra code.